packages/client/src/authentication/__helpers__/setupAuthentication.ts

```diff
@@ -1,41 +1,203 @@
+/* eslint-disable no-console */
 import { Wallet } from 'ethers';
 
 import { mumbaiSandbox } from '../../consts/environments';
+import { Profile } from '../../profile';
+import { getTxIdFromRelayerResult, Transaction } from '../../transaction';
 import { Authentication } from '../Authentication';
-import { setupTestWallet } from './setupTestWallet';
+import { signAndBroadcast } from './signAndBroadcast';
 
 const testConfig = {
   environment: mumbaiSandbox,
 };
 
-export function setupAuthentication() {
-  let auth: Authentication;
+type TestSetup = {
+  authentication: Authentication;
+  profileId: string;
+  wallet: Wallet;
+  walletAddress: string;
+};
+
+type SetupOptions = {
+  withNewProfile?: boolean;
+  withDispatcher?: boolean;
+};
+
+const defaultOptions: SetupOptions = {
+  withNewProfile: false,
+  withDispatcher: false,
+};
+
+const DOES_BURN_PROFILE_WORK = false;
+
+export function setupAuthentication(options?: SetupOptions): () => TestSetup {
+  const { withNewProfile, withDispatcher } = { ...defaultOptions, ...options };
+
+  const testHandle = Date.now().toString();
+  const wallet = Wallet.createRandom();
+
+  const authentication = new Authentication(testConfig);
+  const profile = new Profile(testConfig, authentication);
+  const transaction = new Transaction(testConfig, authentication);
+
+  let _walletAddress: string;
+  let _testProfileId: string;
 
   beforeAll(async () => {
```

Comment:
I found this approach used in some other internal repo testing (e2e specifically). I am not sure one expects `setupAuthentication` to have side effects at the Jest runtime level, but I get what is happening here and why.

Maybe together with this suggestion https://github.com/lens-protocol/lens-sdk/pull/173/files#r1129279335 we can think of something like this?

```ts
describe('Given XYZ', () => {


  describeAuthenticatedScenario(({ }: TestSetup) => {}
  
  })
})
```

where `describeAuthenticatedScenario` also includes a `describe('and the client is authenticated', () => {...})` wrapper so that all relevant test specs includes:

```
Given XYZ
and the client is authenticated
...
```
The `describeAuthenticatedScenario` named after Jest `describe` has the intent of reminding about Jest runtime so maybe the side effect does not appear as a surprise. Hopefully.

Reply:
(`and the client is authenticated` suggested description could well be improvided)

Reply:
that's a very good idea, I will try it!

Reply:
this is the last thing left from the code review, but I'd like to focus on other tasks for SDK v1 release for now.
from what I found out that requires extending jest functions like described here https://spin.atomicobject.com/2020/01/30/jest-add-custom-functions/

Reply:
ok, I added `describeAuthenticatedScenario`, it works like

```ts
describeAuthenticatedScenario({ withNewProfile: true })((getTestSetup) => {
  describe(`when ${Reactions.prototype.add.name} method is called`, () => {
    it(`should execute with success`, async () => {
      const { authentication, profileId } = getTestSetup();
      
      // run test
    });
  });
});
```